test/foundry/deploy/11_HolographFactory.t.sol

```diff
@@ -0,0 +1,353 @@
+
+    vm.expectRevert(bytes(ErrorConstants.INVALID_SIGNATURE_ERROR_MSG));
+    vm.prank(deployer);
+    holographFactory.deployHolographableContract(deployConfig, signature, deployer);
+  }
+  /**
+   * @notice Test the deployHolographableContract function Revert if the signer is invalid
+   * @dev  Refers to the hardhat test with the description 'should fail with invalid signature if signer is incorrect'
+   */
+  function testDeployRevertSignatureSignIncorrect() public {
+    (DeploymentConfig memory deployConfig, bytes32 hashHtokenEth) = getConfigHtokenETH();
+
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashHtokenEth)
+    );
+    Verification memory signature = Verification({v: uint8(bytes1(invalidSignature)), r: r, s: s});
+
+    vm.expectRevert(bytes(ErrorConstants.INVALID_SIGNATURE_ERROR_MSG));
+    vm.prank(deployer);
+    holographFactory.deployHolographableContract(deployConfig, signature, alice);
+  }
+
+  /*
+   * BridgeIn Section
+   */
+  /**
+   * @notice Test the bridgeIn function return the expected selector
+   * @dev  Refers to the hardhat test with the description 'should return the expected selector from the input payload'
+   */
+  function testExpectedSelectorFromPayload() public {
+    vm.skip(true);
+    (DeploymentConfig memory deployConfig, bytes32 hashSampleERC721) = getConfigERC721();
+
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashSampleERC721)
+    );
+    Verification memory signature = Verification({v: uint8(bytes1(invalidSignature)), r: r, s: s});
+
+    bytes memory payload = abi.encode(deployConfig, signature, address(deployer));
+    console.log(block.chainid);
+    console.log(block.chainid);
+    vm.prank(deployer);
+    holographFactory.bridgeIn(uint32(block.chainid), payload);
+  }
+
+  /**
+   * @notice Test the bridgeIn function revert if the payload data is invalid
+   * @dev  Refers to the hardhat test with the description 'should revert if payload data is invalid'
+   */
+  function testRevertDataPayloadInvalid() public {
+    bytes memory payload = "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+    vm.expectRevert();
+    vm.prank(deployer);
+    holographFactory.bridgeIn(uint32(block.chainid), payload);
+  }
+
+  /*
+   * BridgeOut Section
+   */
+  /**
+   * @notice Test the bridgeOut function
+   * @dev  Refers to the hardhat test with the description 'should return selector and payload'
+   */
+  function testContemplateSelectorFromPayload() public {
+    (DeploymentConfig memory deployConfig, bytes32 hashSampleERC721) = getConfigERC721();
+
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashSampleERC721)
+    );
+    Verification memory signature = Verification({v: uint8(bytes1(invalidSignature)), r: r, s: s});
```

Comment:
@mmvazzano this should be using a valid signature.

Reply:
Done!
Same as [here](https://github.com/holographxyz/holograph-protocol/pull/325#discussion_r1626657890)